crawl-ref/source/libutil.cc

```diff
@@ -447,8 +447,11 @@ void set_cursor_region(GotoRegion region)
 
 void clrscr()
 {
-    save_cursor_pos save; // is this generally correct? alternative, set region
-                          // to GOTO_CRT.
+    if (is_cursor_enabled())
+    {
+        save_cursor_pos save; // is this generally correct? alternative, set region
```

Comment:
This won't do what you want; the `save` object is created and goes out of scope within this conditional (so position is never restored after `clrscr_sys`). In general, for conditional RAII you would need to duplicate the calls that the RAII object would go out of scope after, or modify the RAII object to handle conditions itself. Probably the former is easier here.

(I should say I'm not entirely sure if this is the right solution: usually crashes in this object are a symptom rather than a cause. But I haven't done any debugging of this particular case.)

Reply:
Oh my, It worked like a charm in my tests. Let me check it again.